presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ObjectSetCreateBookmarkRecordViewModel.kt

```diff
@@ -4,46 +4,133 @@ import androidx.lifecycle.ViewModel
+import com.anytypeio.anytype.domain.misc.DateProvider
 import com.anytypeio.anytype.domain.objects.CreateBookmarkObject
+import com.anytypeio.anytype.domain.objects.StoreOfRelations
 import com.anytypeio.anytype.domain.workspace.SpaceManager
+import com.anytypeio.anytype.presentation.sets.state.ObjectState
+import com.anytypeio.anytype.presentation.util.Dispatcher
+import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.launch
 import timber.log.Timber
 
 class ObjectSetCreateBookmarkRecordViewModel(
     private val createBookmarkObject: CreateBookmarkObject,
     private val urlValidator: UrlValidator,
-    private val spaceManager: SpaceManager
+    private val spaceManager: SpaceManager,
+    private val objectState: StateFlow<ObjectState>,
+    private val dispatcher: Dispatcher<Payload>,
+    private val addObjectToCollection: AddObjectToCollection,
+    private val session: ObjectSetSession,
+    private val storeOfRelations: StoreOfRelations,
+    private val dateProvider: DateProvider
 ) : SetDataViewObjectNameViewModelBase() {
 
     override fun onActionDone(input: String) {
-        proceedWithCreatingBookmarkObject(input)
+        viewModelScope.launch {
+            proceedWithCreatingBookmarkObject(input)
+        }
     }
 
     override fun onButtonClicked(input: String) {
-        proceedWithCreatingBookmarkObject(input = input)
+        viewModelScope.launch {
+            proceedWithCreatingBookmarkObject(input = input)
+        }
     }
 
-    private fun proceedWithCreatingBookmarkObject(input: String) {
+    private suspend fun proceedWithCreatingBookmarkObject(input: String) {
         if (urlValidator.isValid(url = input)) {
-            viewModelScope.launch {
-                val params = CreateBookmarkObject.Params(
-                    space = spaceManager.get(),
-                    url = input
-                )
-                createBookmarkObject(params).process(
-                    failure = {
-                        Timber.e(it, "Error while creating bookmark object")
-                        sendToast("Error while creating bookmark object")
-                    },
-                    success = { isCompleted.value = true }
-                )
+            val state = objectState.value.dataViewState() ?: return
+            when (state) {
+                is ObjectState.DataView.Collection -> {
+                    val viewer = state.viewerByIdOrFirst(session.currentViewerId.value) ?: return
+                    val prefilled = viewer.prefillNewObjectDetails(
+                        dateProvider = dateProvider,
+                        storeOfRelations = storeOfRelations,
+                        dataViewRelationLinks = state.objectRelationLinks
+                    )
+                    createBookmark(
+                        input = input,
+                        details = prefilled
+                    ) { addBookmarkToCollection(it) }
```

Comment:
please used named parameters instead of using lamba directly - for a better code readability

Reply:
done